### ISSUE TITLE
Type annotation for attr.s class constructor with converter should inherit converter arg type

### ISSUE THREAD

**Issue by V3RGANz** [python-attrs/attrs#792]
Is there a way to define separate types for class constructor argument and attribute?

```python3
@attr.s()
class Example:
    int_attr: int = attr.ib(converter=int)
```

`Example("0")` fails type check, because `"0"` is not int. But if we have `converter` defined, `__init__` argument type should be the same as converter argument type.

Something like:
```python3
def converter(init_attr: Any) -> int:
    ...

def __init__(self, init_attr: Any):  # inherited `Any` from converter
    self.int_attr: int = converter(init_attr)
```

**Comment by hynek** [comment]
If I understand you correctly, you're talking about mypy?

Because our own type hints have fixed this in #710 but the mypy attrs plugin is not something we can affect _here_.

**Comment by V3RGANz** [comment]
Yes, thank you.
